src/gmp.c

```diff
@@ -24815,9 +24815,12 @@ gmp_xml_handle_end_element (/* unused */ GMarkupParseContext* context,
                       setting = g_slist_next (setting);
                     }
 
-                  manage_set_ldap_info (ldap_enabled, ldap_host, ldap_authdn,
-                                        ldap_plaintext, ldap_cacert,
-                                        ldap_ldaps_only);
+                  if (manage_set_ldap_info (ldap_enabled, ldap_host, ldap_authdn,
+                                            ldap_plaintext, ldap_cacert,
+                                            ldap_ldaps_only))
+                    SEND_TO_CLIENT_OR_FAIL (XML_ERROR_SYNTAX
```

Comment:
This will continue around the loop and end at L24854 where it will send a second response. It should have a break like the `SEND_TO_CLIENT_OR_FAIL` above.

Reply:
Hmm, actually the break above it also an error because the OK will also be sent, from what I can see.

The break is supposed to break out of the whole command, but it is just breaking out of the while loop.

Might have to use a flag variable. This case has got a bit long though, would be nice to move the loop body to a function.